modules/vectra.py

```diff
@@ -522,6 +522,124 @@ def delete_rule(self, rule_id=None, restore_detections=True):
         return requests.delete('{url}/rules/{id}'.format(url=self.url, id=rule_id), headers=self.headers, params=params,
                                verify=self.verify)
 
+        @validate_api_v2
+    @request_error_handler
+    def get_group_by_id(self, group_id):
+        """
+        Get groups by id
+        :param rule_id: id of group to retrieve
+        """
+        return requests.get('{url}/groups/{id}'.format(url=self.url, id=group_id), headers=self.headers, verify=False)
+
+    @validate_api_v2
+    def get_all_groups(self):
+        """
+        Get all existing get_all_groups
+        Groups do not implement pagination
+        For consistency's sake, we still implement a generator
+        """
+        resp = self._get_request(url = '{url}/groups'.format(url=self.url))
```

Comment:
Should we handle noo-200 status codes here, and not assume the json returned is interable unless we got a 2xx status code?

Reply:
Well, actually I just realised that I'm being consistent with my forked version of the api tools (which has generators returning JSON and nor requests objects) and not this version. 
I'll submit a fix, especially since here I used the _get_request() function which isn't in upstream version.